DragaliaAPI/Features/Missions/MissionInitialProgressionService.cs

```diff
@@ -14,81 +15,104 @@
+                => await storyRepository.QuestStories.AnyAsync(
+                    x => x.StoryId == progressionInfo.Parameter && x.State == StoryState.Read
+                )
+                    ? 1
+                    : 0,
+            MissionCompleteType.EventGroupCleared
+                => await GetQuestGroupClearedCount(progressionInfo),
+            MissionCompleteType.WeaponEarned => await GetWeaponEarnedCount(progressionInfo),
+            MissionCompleteType.WeaponRefined => await GetWeaponRefinedCount(progressionInfo),
+            MissionCompleteType.AbilityCrestBuildupPlusCount
+                => await GetWyrmprintBuildupCount(progressionInfo),
+            MissionCompleteType.CharacterBuildupPlusCount
+                => await GetCharacterBuildupCount(progressionInfo),
+            MissionCompleteType.PlayerLevelUp
+                => (await userDataRepository.GetUserDataAsync()).Level,
+            MissionCompleteType.AbilityCrestTotalPlusCountUp
+                => (
+                    await abilityCrestRepository.AbilityCrests
+                        .Where(
+                            x =>
+                                progressionInfo.Parameter == null
+                                || x.AbilityCrestId == (AbilityCrests)progressionInfo.Parameter
+                        )
+                        .Select(x => new { x.AttackPlusCount, x.HpPlusCount })
+                        .ToListAsync()
+                )
+                    .Select(x => (int?)Math.Min(x.AttackPlusCount, x.HpPlusCount))
+                    .Max() ?? 0,
+            MissionCompleteType.AbilityCrestLevelUp
+                => (
+                    await abilityCrestRepository.AbilityCrests
+                        .Where(
+                            x =>
+                                progressionInfo.Parameter == null
+                                || x.AbilityCrestId == (AbilityCrests)progressionInfo.Parameter
+                        )
+                        .Select(x => (int?)x.BuildupCount)
+                        .ToListAsync()
+                ).Sum() ?? 0,
+            MissionCompleteType.CharacterLevelUp => await GetCharacterMaxLevel(progressionInfo),
+            MissionCompleteType.CharacterManaNodeUnlock
+                => await GetCharacterManaNodeCount(progressionInfo),
+            MissionCompleteType.DragonLevelUp => await GetDragonMaxLevel(progressionInfo),
+            MissionCompleteType.TreasureTrade => await GetTreasureTradeCount(progressionInfo),
+            MissionCompleteType.DragonBondLevelUp => await GetDragonBondLevel(progressionInfo),
+            MissionCompleteType.PartyPowerReached
+                => await partyPowerRepository.GetMaxPartyPowerAsync(),
+            MissionCompleteType.DragonGiftSent => 0, // Unsure about this
+            MissionCompleteType.ItemSummon => 0, // TODO: As daily quests also use this, should we make it actually count the item summons?
```

Comment:
are there any non-daily endeavours that use this? if it's just daily it's probably ok

Reply:
this is the very first royal regimen quest but other than that and daily I don't think its used